lightstep/resource_user_role_binding.go

```diff
@@ -0,0 +1,188 @@
+package lightstep
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/lightstep/terraform-provider-lightstep/client"
+)
+
+func resourceUserRoleBinding() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceUserRoleCreateOrUpdate,
+		ReadContext:   resourceUserRoleBindingRead,
+		UpdateContext: resourceUserRoleCreateOrUpdate,
+		DeleteContext: resourceUserRoleBindingDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceUserRoleBindingImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"role": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true, // changing role or project requires a new tf resource to ensure permissions are properly removed.
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true, // changing role or project requires a new tf resource to ensure permissions are properly removed.
+			},
+			"users": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func getUserRoleBindingFromResource(_ context.Context, d *schema.ResourceData) client.RoleBinding {
+	var userRoleBinding client.RoleBinding
+
+	role, ok := d.GetOk("role")
+	if ok {
+		userRoleBinding.RoleName = role.(string)
+	}
+
+	project, ok := d.GetOk("project")
+	if ok {
+		userRoleBinding.ProjectName = project.(string)
+	}
+
+	users, ok := d.GetOk("users")
+	if ok {
+		for _, user := range users.(*schema.Set).List() {
+			userRoleBinding.Users = append(userRoleBinding.Users, user.(string))
+		}
+	}
+
+	return userRoleBinding
+}
+
+func setUserRoleBindingFromResource(d *schema.ResourceData, userRoleBinding client.RoleBinding) error {
+	users := schema.NewSet(schema.HashString, nil)
+	for _, user := range userRoleBinding.Users {
+		users.Add(user)
+	}
+
+	err := d.Set("users", users)
+	if err != nil {
+		return fmt.Errorf("error to set users resource field: %v", err)
+	}
+	err = d.Set("role", userRoleBinding.RoleName)
+	if err != nil {
+		return fmt.Errorf("error to set role resource field: %v", err)
+	}
+	err = d.Set("project", userRoleBinding.ProjectName)
+	if err != nil {
+		return fmt.Errorf("error to set project resource field: %v", err)
+	}
+
+	return nil
+}
+
+func resourceUserRoleCreateOrUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
(non blocking suggestion): what do you think about calling this `resourceUserRoleBindingCreateOrUpdate`?